betel/data_set_builder.py

```diff
@@ -0,0 +1,80 @@
+import pathlib
+import abc
+from typing import Dict
+import pandas as pd
+from sklearn import model_selection
+from betel import utils
+from betel import info_files_helpers
+
+
+class DataSetBuilder(metaclass=abc.ABCMeta):
+    """A class for splitting data into train-validation-test sets."""
+
+    _RANDOM_SEED = 2579  # seed for random splitting
+
+    def __init__(self, input_dir: pathlib.Path, storage_dir: pathlib.Path,
+                 split_ratio: (float, float, float) = (0.7, 0.15, 0.15)):
+        """"Constructor.
+
+        :param input_dir: data to be split (output of the scraper,
+        meaning a directory containing the input icons and a csv file
+        describing the whole data set)
+        :param storage_dir: storage directory for split data sets
+        :param split_ratio: the ratio for train-validation-test data sets
+        """
+        self._storage_dir = storage_dir
+        self._storage_dir.mkdir(exist_ok=True, parents=True)
+
+        self._input_dir = input_dir
+        self._input_file = input_dir / utils.SCRAPER_INFO_FILE_NAME
+
+        normalised_split_ratio = (elem / sum(split_ratio) for elem in split_ratio)
```

Comment:
I would default to list comprehensions instead of generator expressions unless you explicitly need a generator expression (since the data would be too large) or it's an anonymous variable.
Generator expressions can only be traversed exactly once, so once you store them in a variable you could accidentally try to traverse it twice leading to subtle bugs